openrtb-core/src/main/protobuf/openrtb.proto

```diff
@@ -1197,6 +1203,78 @@ message BidRequest {
+    // RECOMMENDED by the OpenRTB specification.
+    repeated BrandVersion browsers = 1;
+
+    // A BrandVersion object that identifies the
+    // user agent’s execution platform / OS. Implementers should send a brand
+    // derived from the Sec-CH-UA-Platform header, and version derived
+    // from the Sec-CH-UAPlatform-Version header
+    // or an equivalent JavaScript accessor from NavigatorUAData interface.
+    // This header or accessor are only available for browsers that support User-Agent Client Hints.
+    // RECOMMENDED by the OpenRTB specification.
+    optional BrandVersion platform = 2;
+
+    // 1 if the agent prefers a “mobile” version of the content, if available,
+    // i.e. optimized for small screens or touch input.
+    // 0 if the agent prefers the “desktop” or “full” content.
+    // Implementers should derive this value from the Sec-CH-UAMobile header
+    // or an equivalent JavaScript accessor from NavigatorUAData interface.
+    // This header or accessor are only available for browsers that support User-Agent Client Hints.
+    optional bool mobile = 3;
+
+    // Device’s major binary architecture, e.g. “x86” or “arm”.
+    // Implementers should retrieve this value from the Sec-CH-UA-Arch header
+    // or an equivalent JavaScript accessor from NavigatorUAData interface.
+    // This header or accessor are only available for browsers that support User-Agent Client Hints.
+    optional string architecture = 4;
+
+    // Device’s bitness, e.g. “64” for 64-bit architecture.
+    // Implementers should retrieve this value from the Sec-CH-UA-Bitness header
+    // or an equivalent JavaScript accessor from NavigatorUAData interface.
+    // This header or accessor are only available for browsers that support User-Agent Client Hints.
+    optional string bitness = 5;
+
+    // Device model. Implementers should retrieve this value from the Sec-CH-UAModel header
+    // or an equivalent JavaScript accessor from NavigatorUAData interface.
+    // This header or accessor are only available for browsers that support User-Agent Client Hints.
+    optional string model = 6;
+
+    // The source of data used to create this object, List: User-Agent Source in AdCOM 1.0
+    optional int32 source = 7 [default = 0];
+
+    // Placeholder for vendor specific extensions to this object
+    extensions 100 to 9999;
+  }
+
+  // OpenRTB 2.6: Further identification based on User-Agent Client Hints, the BrandVersion object
+  // is used to identify a device’s browser or similar software component, and the user agent’s
+  // execution platform or operating system.
+  message BrandVersion {
+    // A brand identifier, for example, “Chrome” or “Windows”. The value may be sourced
+    // from the User-Agent Client Hints headers, representing either the
+    // user agent brand (from the Sec-CH-UA-Full-Version header) or
+    // the platform brand (from the Sec-CH-UA-Platform header).
+    required string brand = 1;
```

Comment:
should this not be optional ?
checking against 2.6 public proto

Reply:
I was actually following https://iabtechlab.com/wp-content/uploads/2022/04/OpenRTB-2-6_FINAL.pdf to construct the schema.

Can you point me to the spec you're both looking at?

Reply:
NM, found it: https://developers.google.com/authorized-buyers/rtb/downloads/openrtb-proto

Reply:
you found it yep !
This is actually googles proto file and not straight from IAB

Reply:
well, it's another thing that we stick to in this repo. `required` fields in protobuf is a pain in butt, so it's way much easier to have all fields optional and force them only if/when it's really needed in the app itself. @simontrasler again, correct me if the concept has changed since

Reply:
Oh it makes sense. If it were to become optional afterward, it won't be backward compatible. 
I think we should move to optional too.

Reply:
updated.